### PR TITLE
PR#5867: Explain how to use open_con().

### PR DESCRIPTION
**Pull request by oknet**
The DNS sub-system has 2 work modes:
    
    - Primary-Secondary mode
    - Round-Robin mode
    
    The `open_con()` and `open_cons()` are shared with these 2 modes.
    
    Within the Primary-Secondary mode:
    
    - The `DNSHandler::ip` is used to store the Primary DNS server address.
    - The `validate_ip()` is the API to load the first DNS server address
    into The `DNSHandler::ip`.
    - The `open_con()` is the API to update `DNSHandler::ip` with `target`
    and connect to the address, but you cannot update `DNSHandler::ip` with
    it self.
    
    Within the Round-Robin mode:
    
    - The `DNSHandler::ip` is useless.
    - The `open_con()` is the API to connect to `target`.
    
    The SplitDNS is only available in the Primary-Secondary mode:
    
    - I suspect that the Round-Robin mode is a feature that was added later.
    - The author only made this functionality for the DNS sub-system, and
    forgot to do it for SplitDNS.

### CODE REVIEW

File: iocore/dns/DNS.cc [hunk 1]
```diff
@@ -587,7 +616,7 @@ DNSHandler::startEvent_sdns(int /* event ATS_UNUSED */, Event *e)
   this->validate_ip();
 
   SET_HANDLER(&DNSHandler::mainEvent);
-  open_cons(&ip.sa, false, n_con);
+  open_cons(nullptr, false, n_con);
```
Comment (scw00) [inline iocore/dns/DNS.cc]: I'm ok with this, but giving an explicit param is more clear then hiding it into `open_cons`.

Comment (oknet) [reply iocore/dns/DNS.cc]: Just do the same things as L597-598.

**Comment by scw00** [comment]
[approve ci autest]

File: iocore/dns/DNS.cc [hunk 2]
```diff
@@ -468,13 +468,24 @@ DNSHandler::open_cons(sockaddr const *target, bool failed, int icon)
   Open (and close) connections as necessary and also assures that the
   epoll fd struct is properly updated.
 
+  target == nullptr :
```
[the remark this answers is not in the document]
Comment (oknet) [reply iocore/dns/DNS.cc]: After adding the Round-Robin mode, it is an implicit rule for `open_cons()` and/or `open_con()`.
This is just an internal function of the DNS sub-system. In my opinion, It is not necessary to update to ATS Docs, but I will put it into 'atsinternals`.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**